src/Renci.SshNet/Security/Cryptography/RsaDigitalSignature.cs

```diff
@@ -41,13 +41,20 @@ public RsaDigitalSignature(RsaKey rsaKey, HashAlgorithmName hashAlgorithmName)
         /// </returns>
         protected override byte[] Hash(byte[] input)
         {
-            return _hash.ComputeHash(input);
+#if !NET462
+            using var hash = IncrementalHash.CreateHash(_hashAlgorithmName);
```

Comment:
Would we benefit (performance wise) from creating the hash in the ctor and reusing it here.
Please create a small benchmark to help us decide.

Reply:
In addition to my other comment: this is a correctness fix, not a performance improvement. I would rather work on other things than to run a benchmark for such a change. Sorry for being blunt.

Reply:
So maybe let's move the hash creating to the constructor. What do you think? this is the last change and I will be able to merge it.

Reply:
Makes sense. Done